Promote included plugin builds to public API (#15471)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

<!-- Do not add breaking changes or deprecations here! Add them to the upgrade guide instead. --> 

## Plugin development improvements

### Included plugin builds

> Co-developing a plugin by including its build in settings file was only possible for project plugins.
> Settings plugins had to be developed in isolation, published to a binary repository and then consumed in a project.

Reviewer: ```suggestion
Developing plugins as part of a composite build was so far only possible for project plugins.
Settings plugins always had to be developed in isolation and published to a binary repository.
```

> This release introduces a new API in settings that allows using a settings plugin from an included build:

Reviewer: ```suggestion
This release introduces a new DSL construct in the settings file for including plugin builds.
Build included like that can provide both project and settings plugins.
```

```
pluginManagement {
    includeBuild("../my-settings-plugin")
}
plugins {
    id("my.settings-plugin") 
}
```

> The above example assumes that the included build produces a settings plugin with id `my.settings-plugin`.

Reviewer: ```suggestion
The above example assumes that the included build defines a settings plugin with the id `my.settings-plugin`.
```

> Builds included inside `pluginManagement` block can also contribute project plugins, however, any library components they
> produce will not be visible to the including build.

Reviewer: ```suggestion
Library components produced by builds included though the `pluginManagement` block are not automatically visible to the including build.
```

> If a build produces both plugins and libraries that the including build wants to use, such build should be included
> both in `pluginManagement` block and outside it:

Reviewer: ```suggestion
However, the same build can be included as plugin build and normal library build:
```

```
pluginManagement {
    // contributes plugins
    includeBuild("../project-with-plugin-and-library") 
}
// contributes libraries
includeBuild("../project-with-plugin-and-library") 
```

> This can be viewed similarly to the repository declaration - 

Reviewer: ```suggestion
This distinction reflects what Gradle offers for repository declarations - 
```
